src/main/java/io/appium/java_client/ios/IOSTouchAction.java

```diff
@@ -1,12 +1,28 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
 package io.appium.java_client.ios;
 
 import io.appium.java_client.PerformsTouchActions;
 import io.appium.java_client.TouchAction;
+import io.appium.java_client.ios.touch.DoubleTapOptions;
 import org.openqa.selenium.WebElement;
-import org.openqa.selenium.internal.HasIdentity;
 
 
-public class IOSTouchAction extends TouchAction {
+public class IOSTouchAction extends TouchAction<IOSTouchAction> {
```

Comment:
@mykola-mokhnach Why this class is not fully deprecated?

Reply:
This class will be needed to add further TouchAction extensions